presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPickTableLayout.java

```diff
@@ -182,4 +189,20 @@ public void ruleAddedNewTableLayoutIfTableScanHasEmptyConstraint()
                                         ImmutableMap.of("nationkey", singleValue(BIGINT, 44L)),
                                         ImmutableMap.of("nationkey", "nationkey"))));
     }
+
+    @Test
+    public void ruleAddsFullyEnforcedTableLayout()
```

Comment:
What's a fully enforced predicate?

Reply:
I changed that to `ruleWithPushdownableToTableLayoutPredicate`. Still it is not perfect, but it is better.